Archive/Wiedijk100Theorems/CubingACube.lean

```diff
@@ -415,95 +417,96 @@ variable (h v)
   by_cases h2j₂ : j₂ = j'
-  · cases h2j₂; refine' ⟨x', hx'.1, _⟩; convert hi''.2 j'; simp
+  · cases h2j₂; refine' ⟨x', hx'.1, _⟩; convert hi''.2 j' using 1; simp
   refine' ⟨(cs i).b j₂.succ, _, _⟩
-  · convert hi'.2 j₂; simp [hj₂]
-  · convert hi''.2 j₂; simp [h2j₂]
+  · convert hi'.2 j₂ using 1; simp [hj₂]
+  · convert hi''.2 j₂ using 1; simp [h2j₂]
 #align theorems_100.«82».mi_not_on_boundary Theorems100.«82».mi_not_onBoundary
 
 variable {h v}
 
 /-- The same result that `mi` cannot lie on the boundary of the valley written as inequalities. -/
-theorem mi_not_on_boundary' (j : Fin n) :
+theorem mi_not_onBoundary' (j : Fin n) :
     c.tail.b j < (cs (mi h v)).tail.b j ∧
       (cs (mi h v)).tail.b j + (cs (mi h v)).w < c.tail.b j + c.w := by
-  have := mi_not_on_boundary h v j
-  simp only [on_boundary, not_or] at this ; cases' this with h1 h2
+  have := mi_not_onBoundary h v j
+  simp only [OnBoundary, not_or] at this; cases' this with h1 h2
   constructor
   apply lt_of_le_of_ne (b_le_b mi_mem_bcubes _) h1
   apply lt_of_le_of_ne _ h2
   apply ((Ico_subset_Ico_iff _).mp (tail_sub mi_mem_bcubes j)).2
   simp [hw]
-#align theorems_100.«82».mi_not_on_boundary' Theorems100.«82».mi_not_on_boundary'
+#align theorems_100.«82».mi_not_on_boundary' Theorems100.«82».mi_not_onBoundary'
 
 /-- The top of `mi` gives rise to a new valley, since the neighbouring cubes extend further upward
   than `mi`. -/
 theorem valley_mi : Valley cs (cs (mi h v)).shiftUp := by
   let i := mi h v; have hi : i ∈ bcubes cs c := mi_mem_bcubes
   refine' ⟨_, _, _⟩
-  · intro p; apply h.shift_up_bottom_subset_bottoms mi_xm_ne_one
-  · rintro i' hi' ⟨p2, hp2, h2p2⟩; simp only [head_shift_up] at hi' 
+  · intro p; apply h.shiftUp_bottom_subset_bottoms mi_xm_ne_one
+  · rintro i' hi' ⟨p2, hp2, h2p2⟩; simp only [head_shiftUp] at hi'
     classical
     by_contra h2i'
-    rw [tail_shift_up] at h2p2 
-    simp only [not_subset, tail_shift_up] at h2i' 
+    rw [tail_shiftUp] at h2p2
+    simp only [not_subset, tail_shiftUp] at h2i'
     rcases h2i' with ⟨p1, hp1, h2p1⟩
-    have : ∃ p3, p3 ∈ (cs i').tail.to_set ∧ p3 ∉ (cs i).tail.to_set ∧ p3 ∈ c.tail.to_set := by
-      simp only [to_set, not_forall, mem_set_of_eq] at h2p1 ; cases' h2p1 with j hj
-      rcases Ico_lemma (mi_not_on_boundary' j).1 (by simp [hw]) (mi_not_on_boundary' j).2
+    have : ∃ p3, p3 ∈ (cs i').tail.toSet ∧ p3 ∉ (cs i).tail.toSet ∧ p3 ∈ c.tail.toSet := by
+      simp only [toSet, not_forall, mem_setOf_eq] at h2p1; cases' h2p1 with j hj
+      rcases Ico_lemma (mi_not_onBoundary' j).1 (by simp [hw]) (mi_not_onBoundary' j).2
           (le_trans (hp2 j).1 <| le_of_lt (h2p2 j).2) (le_trans (h2p2 j).1 <| le_of_lt (hp2 j).2)
           ⟨hj, hp1 j⟩ with
         ⟨w, hw, h2w, h3w⟩
       refine' ⟨fun j' => if j' = j then w else p2 j', _, _, _⟩
       · intro j'; by_cases h : j' = j
-        · simp only [if_pos h]; convert h3w
+        · simp only [h, if_pos h]; exact h3w
```

Comment:
This is a regression, no? Why is `h` required in the simp set when it wasn't previously? Looks like more of the same below.

Reply:
In the meantime, I found a cleaner approach: `simp only [if_pos h]; exact h ▸ h3w`. I no longer need `h` in the simp set.

There's no regression in `simp only`. There may be a regression in `convert` but I don't want to debug it.